### PR TITLE
[Android] Removing TapGestureRecognizer with at least 2 taps causes Exception - fix

### LINKED ISSUE #21437: Removing TapGestureRecognizer with at least 2 taps causes Exception
### Description

When tapping a TapGestureRecognizer with NumberOfTapsRequired of at least 2, which will be removed inside the tap command, it will cause a NullReferenceException in [`InnerGestureListener.cs:65`](https://github.com/dotnet/maui/blob/402fdf7fcc5a550ebc32512791a1121d1961c1bb/src/Controls/src/Core/Platform/Android/InnerGestureListener.cs#L65)
Happens on Android emulator and my phones.

### Steps to Reproduce

1. Create a new MAUI App
2. Replace content of MainPage.xaml `<ContentPage>` with

```xaml
<VerticalStackLayout BindableLayout.ItemsSource="{Binding Items}" HorizontalOptions="Start">
    <BindableLayout.ItemTemplate>
        <DataTemplate x:DataType="system:String">
            <Label Text="{Binding .}" HorizontalOptions="Start">
                <Label.GestureRecognizers>
                    <TapGestureRecognizer NumberOfTapsRequired="2" Command="{Binding TapCommand, Source={RelativeSource AncestorType={x:Type local:MainPage}}}" CommandParameter="{Binding .}" />
                </Label.GestureRecognizers>
            </Label>
        </DataTemplate>
    </BindableLayout.ItemTemplate>
</VerticalStackLayout>
```
3. Replace MainPage.xaml.cs with following: 
```csharp
public partial class MainPage : ContentPage
{
    public MainPage()
    {
        InitializeComponent();
        BindingContext = this;

    }
    public ObservableCollection<string> Items { get; } = ["Item 1", "Item 2", "Item 3"];

    public ICommand TapCommand => new Command<string>(obj =>
    {
        Items.Remove(obj);
    });
}
```
4. Run app in Android
5. double-tap Item 1, Item 2 or Item 3

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

Inserting a short `await Task.Delay()` and then removing the entry works. Also going down to 1 required tap works as well. 

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/21437

|Before|After|
|--|--|
|<video src="https://github.com/dotnet/maui/assets/42434498/4b1574cc-0059-4789-b504-4f71a1baeb74" width="300px"/>|<video src="https://github.com/dotnet/maui/assets/42434498/c1740d6e-d7ac-470c-91cf-7cede4634f3e" width="300px"/>|

### CODE REVIEW

**Comment by rmarinho** [comment]
Why does it happen only with more than 2 taps ?

File: src/Controls/tests/UITests/Tests/Issues/Issue21437.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		: base(device)
+	{ }
+
+    [Test]
+	public void ExceptionShouldNotBeThrown()
+	{
+		_ = App.WaitForElement("Item2");
+		App.DoubleClick("Item2");
+
+        //The test passes if no exception is thrown
```
Comment (jsuarezruiz) [inline src/Controls/tests/UITests/Tests/Issues/Issue21437.cs:21]: Could update a Label text double clicking and verify the updated text here; but without crash this is enough.

Comment (kubaflo) [reply src/Controls/tests/UITests/Tests/Issues/Issue21437.cs:21]: Is this test enough then?

**Comment by kubaflo** [comment]
> Why does it happen only with more than 2 taps ?

I suspect it is because of this constructor: https://github.com/dotnet/maui/blob/757f684b8d1d75c13af0dc106361ded8b4fb8425/src/Controls/src/Core/Platform/Android/InnerGestureListener.cs#L68

**Review by jsuarezruiz — DISMISSED** [review]

File: src/Controls/tests/TestCases/Issues/Issue21437.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Controls.Xaml;
+using System.Collections.ObjectModel;
+
+namespace Maui.Controls.Sample.Issues;
+
+[XamlCompilation(XamlCompilationOptions.Compile)]
+[Issue(IssueTracker.Github, 21437, "Removing TapGestureRecognizer with at least 2 taps causes Exception", PlatformAffected.Android)]
+
+public partial class Issue21437 : ContentPage
```
Comment (PureWeen) [inline src/Controls/tests/TestCases/Issues/Issue21437.xaml.cs:10]: Move to new test location

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**